sdk/tables/Azure.Data.Tables/src/TableClient.cs

```diff
@@ -106,60 +135,52 @@ protected TableClient()
+                var (NextPartitionKey, NextRowKey) = ParseContinuationToken(continuationToken);
+
+                var response = _tableOperations.RestClient.QueryEntities(
                     _table,
-                    tableEntityProperties: entity,
-                    queryOptions: new QueryOptions() { Format = _format },
+                    queryOptions: new QueryOptions() { Format = _format, Top = top, Filter = filter, Select = @select, NextPartitionKey = NextPartitionKey, NextRowKey = NextRowKey },
                     cancellationToken: cancellationToken);
-            }
 
-            return response;
+                return Page.FromValues(response.Value.Value,
+                                       CreateContinuationTokenFromHeaders(response.Headers),
+                                       response.GetRawResponse());
+            });
         }
 
-        internal async Task<Response> UpdateInternalAsync(bool async, string partitionKey, string rowKey, IDictionary<string, object> entity, CancellationToken cancellationToken = default)
+        private static string CreateContinuationTokenFromHeaders(TableQueryEntitiesHeaders headers)
         {
-            if (async)
+            if (headers.XMsContinuationNextPartitionKey == null && headers.XMsContinuationNextRowKey == null)
             {
-                return await _tableOperations.UpdateEntityAsync(_table,
-                                                                partitionKey,
-                                                                rowKey,
-                                                                tableEntityProperties: entity,
-                                                                queryOptions: new QueryOptions() { Format = _format },
-                                                                cancellationToken: cancellationToken).ConfigureAwait(false);
+                return null;
             }
             else
             {
-                return _tableOperations.UpdateEntity(_table,
-                                                     partitionKey,
-                                                     rowKey,
-                                                     tableEntityProperties: entity,
-                                                     queryOptions: new QueryOptions() { Format = _format },
-                                                     cancellationToken: cancellationToken);
+                return $"{headers.XMsContinuationNextPartitionKey} {headers.XMsContinuationNextRowKey}";
             }
         }
+
+        private static (string NextPartitionKey, string NextRowKey) ParseContinuationToken(string continuationToken)
+        {
+            // There were no headers passed and the continuation token contains just the space delimiter
+            if (continuationToken?.Length <= 1)
+            {
+                return (null, null);
+            }
+
+            var tokens = continuationToken.Split(' ');
```

Comment:
Are we positive that tokens coming from the service don't have spaces in them?

Reply:
I'm not 100% sure, but it seemed reasonable since the tokens are hashes of the next partition or row key. Do you have any ideas on a better delimiter?

Reply:
Nah, we could start with space and see how that goes.